vowpalwabbit/learner.h

```diff
@@ -673,14 +683,14 @@ struct reduction_learner_builder
     set_params_per_weight(1);
     this->set_learn_returns_prediction(false);
 
-    // By default, will produce what the base produces
-    super::set_output_prediction_type(base->get_output_prediction_type());
-    // By default, will produce what the base produces
-    super::set_input_prediction_type(base->get_output_prediction_type());
-    // By default, will produce what the base expects
-    super::set_output_label_type(base->get_input_label_type());
     // By default, will produce what the base expects
     super::set_input_label_type(base->get_input_label_type());
+    // By default, will produce what the base expects
+    super::set_output_label_type(base->get_input_label_type());
+    // By default, will produce what the base produces
+    super::set_input_prediction_type(base->get_output_prediction_type());
+    // By default, will produce what the base produces
+    super::set_output_prediction_type(base->get_output_prediction_type());
```

Comment:
Whats the reasoning behind this reordering? And the  `set_input_` set_output_` functions above?

Reply:
I wanted to keep the ordering input_label, output_label, input_pred, output_pred throught the code base since it seems easiest to follow